File: frontend/src/api/checklist.ts
import client from './client';
import type {
  ManualChecklistItemInput,
  CheckReportExport,
  SubmissionChecklist,
  SubmissionChecklistInput,
} from '../types';

/**
 * Submission Checklist API module.
 */

/** GET /tenders/{id}/checklist — fetch checklist items (auto-init on first access). */
export async function getChecklist(
  tenderId: number,
): Promise<SubmissionChecklist[]> {
  const res = await client.get<unknown[]>(`/tenders/${tenderId}/checklist`);
  return res.data.map((raw) => normalizeChecklistItem(raw));
}

/** PUT /tenders/{id}/checklist/{itemId} — update a checklist item. */
export async function updateChecklistItem(
  tenderId: number,
  itemId: number,
  data: SubmissionChecklistInput,
): Promise<SubmissionChecklist> {
  const res = await client.put<unknown>(
    `/tenders/${tenderId}/checklist/${itemId}`,
    data,
  );
  return normalizeChecklistItem(res.data);
}

/** POST /tenders/{id}/checklist — manually add a checklist item. */
export async function addManualChecklistItem(
  tenderId: number,
  data: ManualChecklistItemInput,
): Promise<SubmissionChecklist> {
  const res = await client.post<unknown>(
    `/tenders/${tenderId}/checklist`,
    data,
  );
  return normalizeChecklistItem(res.data);
}

/** GET /tenders/{id}/check-report/export — export Markdown check report. */
export async function exportCheckReport(
  tenderId: number,
): Promise<CheckReportExport> {
  const res = await client.get<CheckReportExport>(
    `/tenders/${tenderId}/check-report/export`,
  );
  return res.data;
}

/** Normalize raw API response to SubmissionChecklist. */
function normalizeChecklistItem(raw: unknown): SubmissionChecklist {
  const item = raw as Record<string, unknown>;
  const status = item.status;
  return {
    id: Number(item.id),
    tender_id: Number(item.tender_id),
    requirement_id:
      item.requirement_id == null ? null : Number(item.requirement_id),
    item_name: String(item.item_name || ''),
    description:
      item.description == null ? null : String(item.description),
    status:
      status === 'in_progress' || status === 'done'
        ? status
        : 'not_started',
    remark: item.remark == null ? null : String(item.remark),
    created_at: String(item.created_at || ''),
    updated_at: String(item.updated_at || ''),
  };
}

File: backend/app/models/checklist.py
"""投标待办清单数据模型。

对应 SQLite 表 submission_checklist，用于 submission 类要求的
三档状态（未开始/进行中/已完成）管理。
"""

from typing import Optional

from pydantic import BaseModel


class SubmissionChecklist(BaseModel):
    """待办清单模型，对应 submission_checklist 表。

    Attributes:
        id: 主键，自增。
        tender_id: 外键，关联 tenders.id。
        requirement_id: 外键，关联 tender_requirements.id（手动新增的项可为 None）。
        item_name: 待办项名称。
        description: 详细说明。
        status: 状态（not_started / in_progress / done）。
        remark: 备注（灰色地带提示等）。
        created_at: 记录创建时间（ISO 8601）。
        updated_at: 记录更新时间（ISO 8601）。
    """

    id: Optional[int] = None
    tender_id: int = 0
    requirement_id: Optional[int] = None
    item_name: str = ""
    description: Optional[str] = None
    status: str = "not_started"
    remark: Optional[str] = None
    created_at: str = ""
    updated_at: str = ""
